File: GenomicRangeQuery.cpp
// you can use includes, for example:
// #include <algorithm>

// you can write to stdout for debugging purposes, e.g.
// cout << "this is a debug message" << endl;

vector<int> solution(string &S, vector<int> &P, vector<int> &Q) {
    // write your code in C++11 (g++ 4.8.2)
    vector<vector<int>> vACGT(4, vector<int>{0});
    vector<int> res;
    auto len = S.size();
    int a,c,g,t;
    a=c=g=t=0;
    for (unsigned int i=0; i<len; i++)
    {
        switch (S[i]){
            case 'A':
                a++;
                break;
            case 'C':
                c++;
                break;
            case 'G':
                g++;
                break;
            case 'T':
                t++;
                break;
            default:
                break;
        }
        vACGT[0].push_back(a);    
        vACGT[1].push_back(c);
        vACGT[2].push_back(g);
        vACGT[3].push_back(t);
    }
    
    //Caculate the result
    for (unsigned int m=0; m<P.size(); m++)
    {
        for (unsigned int n=0; n<4; n++)
        {
            if (vACGT[n][Q[m]+1] - vACGT[n][P[m]] > 0)
            {
                res.push_back(n+1);
                break;
            }
        }
    }
    return res;
}

File: Prefix Sums/GenomicRangeQuery.py
# you can write to stdout for debugging purposes, e.g.
# print "this is a debug message"

def solution(S, P, Q):
    # write your code in Python 2.7
    le = len(S)
    lens = le + 1
    Sum=[[0]*lens,[0]*lens,[0]*lens, [0]*lens]

    a=c=g=t=0
    for i,nuc in enumerate(S):
        if nuc == "A":
            a+=1
        elif nuc == "C":
            c+=1
        elif nuc == "G":
            g+=1
        elif nuc == "T":
            t+=1
        Sum[0][i+1]=a
        Sum[1][i+1]=c
        Sum[2][i+1]=g
        Sum[3][i+1]=t

    #Caculate result
    result = []
    #print "Sum: ",Sum
    #print "Len(P)", len(P)
    for i in range(len(P)):
        #print "i: ",i,P[i], Q[i]
        for j in range(0,4):
            if Sum[j][Q[i]+1] - Sum[j][P[i]] > 0:
                result.append(j+1)
                break

    #print "Result: ",result
    return result    